envoy/http/header_validator.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include "envoy/config/typed_config.h"
+#include "envoy/http/header_map.h"
+#include "envoy/server/factory_context.h"
+
+namespace Envoy {
+namespace Http {
+
+/**
+ * Interface for header key formatters that are stateful. A formatter is created during decoding
+ * headers, attached to the header map, and can then be used during encoding for reverse
+ * translations if applicable.
+ */
+class HeaderValidator {
+public:
+  virtual ~HeaderValidator() = default;
+
+  enum class HeaderEntryValidationResult { Accept, Reject };
+
+  /**
+   * Method for validating a header entry. This is called for both request and response headers.
+   * Returning the Reject value causes the request to be rejected with the 400 status.
+   */
```

Comment:
How is this going to set RC details?  will the Validator be created with the stream info or should we return details here as well?

here and below

Reply:
I've added plumbing for StreamInfo.